File: Merge two BST 's.py
class Solution:
    def inorder(self, root, ans):
        # If the current node is None, return
        if root is None:
            return 
        # Recursively traverse the left subtree
        self.inorder(root.left, ans)
        # Add the current node's data to the answer list
        ans.append(root.data)
        # Recursively traverse the right subtree
        self.inorder(root.right, ans)

    def merge(self, root1, root2):
        # Initialize an empty list to hold the inorder traversal results
        ans = []
        # Perform inorder traversal on the first tree and store the results in ans
        self.inorder(root1, ans)
        # Perform inorder traversal on the second tree and store the results in ans
        self.inorder(root2, ans)
        # Sort the combined results from both trees
        ans.sort()
        # Return the sorted list
        return ans

File: Merge two BST 's.java
// User function Template for Java

/*
class Node
{
    int data;
    Node left, right;

    public Node(int d)
    {
        data = d;
        left = right = null;
    }
}

*/
class Solution {
    // Function to return a list of integers denoting the node
    // values of both the BST in a sorted order.
    public List<Integer> merge(Node root1, Node root2) {
        // Write your code here
        List<Integer> list = new ArrayList<>();
        
        inorder(root1, list);
        inorder(root2, list);
        Collections.sort(list);
        return list;
        
    }
    public static void inorder(Node root, List<Integer> list){
        if(root==null) return;
        inorder(root.left, list);
        list.add(root.data);
        inorder(root.right, list);
    }
    
}
